src/analyses/baseInvariant.ml

```diff
@@ -83,7 +83,11 @@ struct
       in
       let state_with_excluded = set a gs st addr t_lval value ~ctx in
       let value =  get a gs state_with_excluded addr None in
-      let new_val = apply_invariant oldval value in
+      let new_val =
+        try
+          apply_invariant oldval value
+        with PreValueDomain.NotMeetable -> oldval
```

Comment:
Seems to me like `Lattice.Uncomparable` already exists for this precise meaning:
https://github.com/goblint/analyzer/blob/3af00a86da65ab5caac033b0ecff02dc69c3bfbf/src/domains/lattice.ml#L14-L15
Just using that exception would avoid the need for littering interfaces with some special `try_meet` because `meet` is allowed to raise `Uncomparable`.

Then `try_meet` could just be a helper function here which has such fallback. Right now that code is repeated multiple times.

Reply:
Yes, that is much more concise indeed. I adapted the implementation accordingly.